File: packages/types/src/tenant.ts
export type TemplateType = 'HOSPITAL' | 'PENSION' | 'STARTUP' | 'GENERAL'

export type PlanType = 'BASIC' | 'STANDARD' | 'PREMIUM'

export interface Tenant {
  id: string
  slug: string
  name: string
  template_type: TemplateType
  plan_type: PlanType
  plan_expires_at: string | null
  custom_domain: string | null
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface TenantCreate {
  slug: string
  name: string
  template_type: TemplateType
  plan_type?: PlanType
}

export interface TenantUpdate {
  name?: string
  template_type?: TemplateType
  plan_type?: PlanType
  plan_expires_at?: string | null
  custom_domain?: string | null
  is_active?: boolean
}

File: apps/backend/app/models/tenant.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base, SoftDeleteMixin, TimestampMixin


class Tenant(Base, TimestampMixin, SoftDeleteMixin):
    __tablename__ = "tenants"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    slug: Mapped[str] = mapped_column(String(100), unique=True, nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    template_type: Mapped[str] = mapped_column(String(50), nullable=False, default="GENERAL")
    plan_type: Mapped[str] = mapped_column(String(50), nullable=False, default="BASIC")
    plan_expires_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    custom_domain: Mapped[str | None] = mapped_column(String(255), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)

    users: Mapped[list["User"]] = relationship("User", back_populates="tenant", lazy="noload")
    sections: Mapped[list["Section"]] = relationship("Section", back_populates="tenant", lazy="noload")
